akka-actor/src/main/scala/akka/dispatch/Mailbox.scala

```diff
@@ -385,6 +385,9 @@ trait MessageQueue {
    * is expected to transfer all remaining messages into the dead letter queue
    * which is passed in. The owner of this MessageQueue is passed in if
    * available (e.g. for creating DeadLetters()), “/deadletters” otherwise.
+   *
+   * Note that we implement the method in a recursive manner mainly for
+   * atomicity (not touching the queue twice).
```

Comment:
This shouldn't go in the API doc but as a code comment. A caller does not need know this and it is not a requirement by the API. I see that we have several identical impls of it though, so I think it is fine to put the non API-comment here even though this is the interface and not the impl.

Reply:
Probably a misunderstanding of what needs to be done here. I'm merging this anyway.